scio-parquet/src/test/scala/com/spotify/scio/parquet/avro/ParquetAvroTest.scala

```diff
@@ -30,7 +30,7 @@ import org.scalatest._
 object ParquetAvroJob {
   def main(cmdlineArgs: Array[String]): Unit = {
     val (sc, args) = ContextAndArgs(cmdlineArgs)
-    sc.parquetAvroFile[TestRecord](args("input"))
+    sc.parquetAvroFile[TestRecord](args("input")).map(identity)
```

Comment:
What is the purpose of this extra `map`?

Reply:
Because it needs 2 implicit conversions to go from `ParquetAvroFile` to `SCollection` to `ParquetSCollection`. I'll add implicit methods in a companion `object ParquetAvroFile` to make it possible. It may not cover some other cases though, so we won't be able to disguise this completely as a regular `SCollection`.

Reply:
Ah - i see now. What do you thinkg about having it as an argument in `parquetAvroFile` instead of having intermediate type. So it would be:

```
     def parquetAvroFile[T <: SpecificRecordBase : ClassTag, U: ClassTag](path: String,
                                                              projection: Schema = null,
                                                              predicate: FilterPredicate = null,
                                                              extraction: (T => U))
```

or curried version:

```
     def parquetAvroFile[T <: SpecificRecordBase : ClassTag, U: ClassTag](path: String,
                                                              projection: Schema = null,
                                                              predicate: FilterPredicate = null)
                                                              (extraction: (T => U))
```